src/System.Text.Encodings.Web/tests/JavaScriptStringEncoderTests.cs

```diff
@@ -18,11 +19,29 @@ public partial class JavaScriptStringEncoderTests
         public void TestSurrogate()
         {
             Assert.Equal("\\uD83D\\uDCA9", System.Text.Encodings.Web.JavaScriptEncoder.Default.Encode("\U0001f4a9"));
+
             using (var writer = new StringWriter())
             {
                 System.Text.Encodings.Web.JavaScriptEncoder.Default.Encode(writer, "\U0001f4a9");
                 Assert.Equal("\\uD83D\\uDCA9", writer.GetStringBuilder().ToString());
             }
+
+            Span<char> destination = new char[12];
+            OperationStatus status = System.Text.Encodings.Web.JavaScriptEncoder.Default.Encode("\U0001f4a9".AsSpan(), destination, out int charsConsumed, out int charsWritten, isFinalBlock: true);
```

Comment:
At some point, we should probably fix up the namespace of these tests (rename `Microsoft.Framework.WebEncoders` to System.Text.Encodings.Web.Tests`) (outside of this PR, ofc).

Reply:
https://github.com/dotnet/corefx/issues/40073